test/TestCaseReader.h

```diff
@@ -31,6 +32,7 @@ namespace solidity::frontend::test
 struct SourceMap
 {
 	std::map<std::string, std::string> sources;
+	std::vector<std::pair<std::string, boost::filesystem::path>> externalSources;
```

Comment:
```suggestion
	std::vector<std::tuple<std::string, boost::filesystem::path>> externalSources;
```
Why not a map though?

Reply:
I thought that it may be important to preserve the original `ExternalSource` order. So that when `isoltest` is updating the expectations, we will have the original order again.

Reply:
Well, I think the order might affect which error messages you get or the AST IDs but I think it should not change the execution results. I agree that it might not hurt to keep the order just in case but we already ignore it in `sources` so I'd suggest ignoring it here too for simplicity.

If you ignore i and use a map, it will be easier to check if you do not have duplicates.

Reply:
Ok, I changed that to a map. But I think we don't need to check `externalSources` for duplicates here, because the `source` map also contains external sources.